### PR TITLE
Implemented explosives

### PR DESCRIPTION
**Pull request by Fontbane**
- Implemented Damp, and "explosive" move attribute
- Implemented Mind Blown and Steel Beam (recoil taken even if move doesn't hit) with HalfHpAttackAttr
- Implemented Hydro Steam, Misty Explosion and Spirit Shackle
- Status immunity abilities that cure the status when active now do so (for example, a paralyzed mon gaining Limber, or a mon with Mycelium Might using Spore on a mon with Insomnia) via ContinuousResetStatusAbAttr. Note that I've marked Pastel Veil as (P), because it's also supposed to apply to its allies. This can probably be implemented in conjunction with Healer.
- Fixed a console error due to lacking null check in Quick Feet condition

I have not yet tested this yet.

### CODE REVIEW

File: src/data/move.ts [hunk 1]
```diff
@@ -1144,6 +1165,16 @@ export class ClearTerrainAttr extends MoveEffectAttr {
   }
 }
 
+export class Explosive extends MoveAttr {
```
Comment (Flashfyre) [inline src/data/move.ts:1168]: Per convention, always end Attr classes with Attr

**Review by Flashfyre — CHANGES_REQUESTED** [review]

File: src/data/move.ts (hunk 1 above)
Comment (Fontbane) [reply src/data/move.ts]: Fixed this and also cleaned up ContinuousResetStatusAbAttr a little

File: src/field/pokemon.ts [hunk 2]
```diff
@@ -1185,7 +1185,8 @@ export default abstract class Pokemon extends Phaser.GameObjects.Container {
         else {
           if (source.findTag(t => t instanceof TypeBoostTag && (t as TypeBoostTag).boostedType === type))
             power.value *= 1.5;
-          const arenaAttackTypeMultiplier = this.scene.arena.getAttackTypeMultiplier(type, source.isGrounded());
+          const arenaAttackTypeMultiplier = (move.id == Moves.HYDRO_STEAM && this.scene.arena.weather.weatherType==WeatherType.SUNNY)?
```
Comment (Flashfyre) [inline src/field/pokemon.ts:1188]: Is there a better way to do this without hardcoding?

**Review by Flashfyre — CHANGES_REQUESTED** [review]

File: src/field/pokemon.ts (hunk 2 above)
Comment (Fontbane) [reply src/field/pokemon.ts]: Added IgnoreWeatherDebuffsAttr

**Comment by Fontbane** [comment]
The mid-turn status healing didn't work as expected, so for now I've implemented a temporary solution (heal on switch out or end of turn). It'll be easier to implement if we get some kind of "PostAbilityGainAbAttr"

**Pull request closed without merging**